File: Ticket_Raising_Backend/TicketClassLibrary/Models/TicketType.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TicketClassLibrary.Models;

[Table("TicketType")]
public class TicketType
{
    [Key]
    [Column(TypeName = "char(5)")]
    [Required(ErrorMessage = "Ticket Type ID is required.")]
    [StringLength(5, MinimumLength = 5, ErrorMessage = "Ticket Type ID must be exactly 5 characters.")]
    public string TicketTypeId { get; set; }

    [Column(TypeName = "varchar(50)")]
    [Required(ErrorMessage = "Type name is required.")]
    [StringLength(50, ErrorMessage = "Type name cannot exceed 50 characters.")]
    public string TypeName { get; set; }

    [Column(TypeName = "varchar(100)")]
    [StringLength(100, ErrorMessage = "Description cannot exceed 100 characters.")]
    public string Description { get; set; }

    [ForeignKey(nameof(TicketPriority))]
    [Column(TypeName = "char(5)")]
    [Required(ErrorMessage = "Priority ID is required.")]
    [StringLength(5, MinimumLength = 5, ErrorMessage = "Priority ID must be exactly 5 characters.")]
    public string PriorityId { get; set; }

    public virtual TicketPriority? TicketPriority { get; set; }
    public virtual ICollection<Ticket> Tickets { get; set; } = new List<Ticket>();
}

File: Ticket_Raising_Frontend/Ticket_Raising_App/src/Models/TicketType.ts
export class TicketType {
    public ticketTypeId: string;
    public typeName: string;
    public description: string;
    public priorityId: string;
 
    constructor(
        ticketTypeId: string = "",
        typeName: string = "",
        description: string = "",
        priorityId: string = ""
    ) {
        this.ticketTypeId = ticketTypeId;
        this.typeName = typeName;
        this.description = description;
        this.priorityId = priorityId;
    }
}